routers/admin/users.go

```diff
@@ -259,6 +259,14 @@ func EditUserPost(ctx *context.Context, form auth.AdminEditUserForm) {
 		u.HashPassword(form.Password)
 	}
 
+	if u.IsLocal() && len(form.UserName) > 0 && u.Name != form.UserName {
```

Comment:
"...I think `models.ChangeUserName` and `models.UpdateUser` should be with a transaction....."

Reply:
@lunny this would be a refactor witch should be also done on the user-settings and org-settings code ...

https://github.com/go-gitea/gitea/blob/9cc5a89b8c8654e3e40686323572e69300807629/routers/org/setting.go#L63
https://github.com/go-gitea/gitea/blob/9cc5a89b8c8654e3e40686323572e69300807629/routers/user/setting/profile.go#L41